pkg/services/container/server.go

```diff
@@ -98,9 +98,13 @@ func (s *server) getVerifiedSessionToken(req interface {
 		return nil, fmt.Errorf("decode: %w", err)
 	}
 
-	if !token.VerifySignature() {
+	signer, err := icrypto.VerifyTokenSignature(&token)
+	if err != nil {
 		return nil, errors.New("invalid signature")
 	}
+	if signer != token.Issuer() {
```

Comment:
Like here, this check can be embedded into `VerifyTokenSignature()`.

Reply:
done